src/lf_apply.rs

```diff
@@ -389,34 +400,31 @@ unsafe fn filter_plane_cols_y<BD: BitDepth>(
 
 #[inline]
 unsafe fn filter_plane_rows_y<BD: BitDepth>(
-    f: *const Rav1dFrameContext,
-    have_top: c_int,
+    f: &Rav1dFrameContext,
+    have_top: bool,
     lvl: &[[u8; 4]],
     b4_stride: ptrdiff_t,
-    mask: *const [[u16; 2]; 3],
+    mask: &[[[u16; 2]; 3]; 32],
     mut dst: *mut BD::Pixel,
     ls: ptrdiff_t,
     w: c_int,
     starty4: c_int,
     endy4: c_int,
 ) {
-    let dsp: *const Rav1dDSPContext = (*f).dsp;
+    let dsp: &Rav1dDSPContext = &*(*f).dsp;
     for (y, lvl) in (starty4..endy4).zip(lvl.chunks(b4_stride as usize)) {
-        if !(have_top == 0 && y == 0) {
+        if !(!have_top && y == 0) {
             let vmask: [u32; 4] = [
-                (*mask.offset(y as isize))[0][0] as c_uint
-                    | ((*mask.offset(y as isize))[0][1] as c_uint) << 16,
-                (*mask.offset(y as isize))[1][0] as c_uint
-                    | ((*mask.offset(y as isize))[1][1] as c_uint) << 16,
-                (*mask.offset(y as isize))[2][0] as c_uint
-                    | ((*mask.offset(y as isize))[2][1] as c_uint) << 16,
-                0 as c_int as u32,
+                mask[y as usize][0][0] as u32 | (mask[y as usize][0][1] as u32) << 16,
+                mask[y as usize][1][0] as u32 | (mask[y as usize][1][1] as u32) << 16,
+                mask[y as usize][2][0] as u32 | (mask[y as usize][2][1] as u32) << 16,
+                0,
             ];
             (*dsp).lf.loop_filter_sb[0][1](
                 dst.cast(),
                 ls,
                 vmask.as_ptr(),
-                lvl[0][1..].as_ptr() as *const [u8; 4],
+                &lvl[0][1] as *const u8 as *const [u8; 4],
```

Comment:
See #731 for a fix of this.  It's also done as an "unaligned" slice rather than a single "unaligned" read now, as the `fn` ptr reads multiple levels not just one.

Reply:
Fixed in #731.